techniques/system/distributePolicy/1.0/propagatePromises.st

```diff
@@ -45,6 +47,17 @@ bundle agent propagatePromises
     root_server::
       "rudder_tools_updated_exists" expression => fileexists("${g.rudder_tools_updated_origin}");
 
+    !root_server::
+       "sync_method_disable" expression => strcmp("&RELAY_SYNC_METHOD&", "disable");
+       "sync_method_rsync" expression => strcmp("&RELAY_SYNC_METHOD&", "rsync");
+       "sync_promises" expression => strcmp("&RELAY_SYNC_PROMISES&", "true");
+       "sync_sharedfiles" expression => strcmp("&RELAY_SYNC_SHAREDFILES&", "true");
+
+       "disable_classic_promises" expression => "sync_promises.(sync_method_rsync|sync_method_disable)";
```

Comment:
why sync_promises here?

Reply:
We disable classic synchronization of promises when other synchronization method of the promises directory is asked for.

Reply:
i'm not sure I understand - it should be disabled if sync_method_disable or sync_method_rsync; whatever the rest is

Reply:
Yes but if RELAY_SYNC_PROMISES is false and sync is disabled, we want to disable classic promises too?

Reply:
RELAY_SYNC_PROMISES can only be set in rsync mode, so we should ignore it in classic mode (at least, that's what the spec said)

Reply:
yes we should ignore it in classic mode, so if it set and in classic mode we don't disable it (just in case)